cms/static/js/views/video_thumbnail.js

```diff
@@ -0,0 +1,221 @@
+                    'showHoverState', 'hideHoverState'
+                );
+            },
+
+            render: function() {
+                HtmlUtils.setHtml(
+                    this.$el,
+                    this.template({
+                        action: this.action,
+                        imageAltText: this.getImageAltText(),
+                        videoId: this.model.get('edx_video_id'),
+                        actionInfo: this.actionsInfo[this.action],
+                        thumbnailURL: this.model.get('course_video_image_url'),
+                        duration: this.getDuration(this.model.get('duration'))
+                    })
+                );
+                this.hideHoverState();
+                return this;
+            },
+
+            getImageAltText: function() {
+                return StringUtils.interpolate(
+                    // Translators: message will be like Thumbnail for Arrow.mp4
+                    gettext('Thumbnail for {videoName}'),
+                    {videoName: this.model.get('client_video_id')}
+                );
+            },
+
+            getSRText: function() {
+                return StringUtils.interpolate(
+                    // Translators: message will be like Add Thumbnail - Arrow.mp4
+                    gettext('Add Thumbnail - {videoName}'),
+                    {videoName: this.model.get('client_video_id')}
+                );
+            },
+
+            getDuration: function(durationSeconds) {
+                if (durationSeconds <= 0) {
+                    return null;
+                }
+
+                return {
+                    humanize: this.getDurationTextHuman(durationSeconds),
+                    machine: this.getDurationTextMachine(durationSeconds)
+                };
+            },
+
+            getDurationTextHuman: function(durationSeconds) {
+                var humanize = this.getHumanizeDuration(durationSeconds);
+
+                // This case is specifically to handle values between 0 and 1 seconds excluding upper bound
+                if (humanize.length === 0) {
+                    return '';
+                }
+
+                return StringUtils.interpolate(
+                    // Translators: humanizeDuration will be like 10 minutes, an hour and 20 minutes etc
+                    gettext('Video duration is {humanizeDuration}'),
+                    {
+                        humanizeDuration: humanize
+                    }
+                );
+            },
+
+            getHumanizeDuration: function(durationSeconds) {
+                var minutes,
+                    seconds,
+                    minutesText = null,
+                    secondsText = null;
+
+                minutes = Math.trunc(moment.duration(durationSeconds, 'seconds').asMinutes());
+                seconds = moment.duration(durationSeconds, 'seconds').seconds();
```

Comment:
Wouldn't this calculate all seconds ? we need to add only the remaining seconds that are left out after minutes calculations